### ISSUE TITLE
Unhandled exception

### ISSUE THREAD

**Issue by soikon** [digitalsleuth/peepdf-3#17]
Hi, can you check for this error. Seems like the function is still using syntax from Python 2 (I'm not sure though)
```python
Traceback (most recent call last):
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\peepdf.py", line 280, in main
    ret, pdf = pdfParser.parse(
               ^^^^^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCore.py", line 7773, in parse
    ret = pdfFile.decrypt()
          ^^^^^^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCore.py", line 5821, in decrypt
    elif isOwnerPass(
         ^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCrypto.py", line 328, in isOwnerPass
    newKey += chr(ord(rc4Key[i]) ^ counter)
                  ^^^^^^^^^^^^^^
TypeError: ord() expected string of length 1, but int found
``` 

Thank you!

**Comment by digitalsleuth** [comment]
You are correct, this is something I missed during the port to Py 3. I've got it fixed now and will be testing it shortly. Are you able to share the file you were trying it on so I can ensure everything works as expected?

**Comment by soikon** [comment]
Ah I see, unfortunately the sample file contains PII I can't share. Will test once it's released.

**Comment by digitalsleuth** [comment]
Okay, you can give it a shot by following these steps:
```bash
python3 -m pip install --extra-index-url https://test.pypi.org/simple peepdf-3==3.1.0rc1
```
Keep in mind, you may need to use sudo for the command if that's the way you typically install your applications, and have previously installed peepdf-3.

**Comment by soikon** [comment]
Hi, sorry for late reply. I was locked out of my Github account and just got it recovered. Will get back to you within 12h.

**Comment by soikon** [comment]
```python
Traceback (most recent call last):
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\peepdf.py", line 280, in main
    ret, pdf = pdfParser.parse(
               ^^^^^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCore.py", line 7773, in parse
    ret = pdfFile.decrypt()
          ^^^^^^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCore.py", line 5821, in decrypt
    elif isOwnerPass(
         ^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCrypto.py", line 330, in isOwnerPass
    dictO = RC4(dictO, newKey)
            ^^^^^^^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCrypto.py", line 359, in RC4
    y = (y + int(box[x]) + int(hash[x])) % 256
                           ^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'I'

``` 

Getting this error now.

**Comment by digitalsleuth** [comment]
Thanks for getting back to me. I'll also take a look right now to see what might be causing it on this end.

**Comment by soikon** [comment]
Hi, I just looked for the original source code:
https://github.com/jesparza/peepdf/blob/master/PDFCrypto.py

```python
#Initialization
    for x in range(256):
        hash[x] = ord(key[x % keyLength])
        box[x] = x
    for x in range(256):
        y = (y + int(box[x]) + int(hash[x])) % 256
        tmp = box[x]
        box[x] = box[y]
        box[y] = tmp 
```

and in peepdf-3:
```python
    # Initialization
    for x in range(256):
        hash[x] = key[x % keyLength]
        box[x] = x
    for x in range(256):
        y = (y + int(box[x]) + int(hash[x])) % 256
        tmp = box[x]
        box[x] = box[y]
        box[y] = tmp
```

There is a slight ord() function difference.

However, when I change the PDFCrypto.py file to the original ones, I get this error:
```python
Traceback (most recent call last):
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\peepdf.py", line 280, in main
    ret, pdf = pdfParser.parse(
               ^^^^^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCore.py", line 7773, in parse
    ret = pdfFile.decrypt()
          ^^^^^^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCore.py", line 5821, in decrypt
    elif isOwnerPass(
         ^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCrypto.py", line 330, in isOwnerPass
    dictO = RC4(dictO, newKey)
            ^^^^^^^^^^^^^^^^^^
  File "C:\Users\[username]\AppData\Local\Programs\Python\Python312\Lib\site-packages\peepdf\PDFCrypto.py", line 372, in RC4
    ret += chr(ord(data[x]) ^ k)
               ^^^^^^^^^^^^
TypeError: ord() expected string of length 1, but int found
```

Looks strange, will have a look in a few hours because I've got something else to do.

**Comment by digitalsleuth** [comment]
Yes, the original code, based on Python 2, uses ord() to get the integer value of the character provided, but this is no longer required in Python 3, so it's been removed. The values of `hash` and `box` start off as an empty dict in this function, and the `key` value is a byte value. The function populates the two dicts with integer values.

For some reason, the value of x in that function is not being converted to an `int` when it should be from the first attempt you took. The second attempt (when using the original code) would have been expected to fail given the conversion using `ord` and `chr` together in Python 3. 

Since you can't share the file itself, it would be useful if, during your review of this issue, you could identify the `type` for both the `data` value and `key` value, and review their content to see if indeed they are byte values. Please let me know how this goes when you can.

Cheers

**Comment by ekoh9704** [comment]
Hi, I have ran into the same few errors as user soikon mentioned above while I was testing a pdf file from MalwareBazaar and noticed this conversation is still open.
Here is the link to the file I was testing
https://bazaar.abuse.ch/sample/6806728dcfb92c5e551fd0db5e4e35472be44443c502e1827d2943428c2267b6/
Do be careful when testing with this file as it is a malware sample and most likely malicious.
Hopefully this file helps with your testing and allow you to resolve this issue. Do lmk if you need more information. Thanks!

**Comment by digitalsleuth** [comment]
Hi @ekoh9704 , thanks for reaching out. I realize I forgot to push my updates to this which would help identify and fix the issue. I'll take a look at the code again this weekend and push out a new release.

Cheers!

**Comment by digitalsleuth** [comment]
Hi @ekoh9704, This issue has been identified and has been resolved in the next upcoming release. I'm currently doing some linting and will be releasing this in the next 24 hours.

Cheers!

**Comment by digitalsleuth** [comment]
Hi @soikon and @ekoh9704 , these issues are fixed in the latest release, v4.0.0.
Cheers!

**Comment by digitalsleuth** [comment]
Haven't received any updates on this. If the issue still exists, please open a new issue.
